File: programmers/코딩테스트 고득점 Kit/정렬/H-Index.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<int> citations) {
	int answer = 0;
	sort(citations.begin(), citations.end());
	for (int i = 0; i <= 1000; i++) {
		int cnt = 0;
		for (int j = 0; j < citations.size(); j++) {
			if (citations[j] >= i) cnt++;
			if (cnt == i) {
				answer = max(answer, cnt);
				break;
			}
		}
	}

	return answer;
}

File: programmers/코딩테스트 고득점 Kit/정렬/H-index.java
import java.util.Arrays;
class Solution {
    public int solution(int[] citations) {
        int n = citations.length;
        int hIndex = 0;
        Arrays.sort(citations);
        for(int h=0;h<=1000;h++) {
            for(int i=0;i<citations.length;i++) {
                int num = citations[i];
                if(num >= h && citations.length - i >= h) {
                    hIndex = h;
                    break;
                }
            }
        }
        return hIndex;
    }
}